File: Serenity.Script.UI/Editor/IntegerEditor.cs
using jQueryApi;
using System;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serenity
{
    [Editor, DisplayName("Tamsayı"), OptionsType(typeof(IntegerEditorOptions))]
    [Imported(ObeysTypeSystem = true), Element("<input type=\"text\"/>")]
    public class IntegerEditor : Widget<IntegerEditorOptions>
    {
        static IntegerEditor()
        {
            Q.Prop(typeof(IntegerEditor), "value");
        }

        public IntegerEditor(jQueryObject input, IntegerEditorOptions opt)
            : base(input, opt)
        {
            input.AddClass("integerQ");

            dynamic numericOptions = jQuery.ExtendObject(DecimalEditor.DefaultAutoNumericOptions(), new
            {
                vMin = options.MinValue ?? 0,
                vMax = options.MaxValue ?? 2147483647,
                aSep = (string)null
            });

            ((dynamic)input).autoNumeric(numericOptions);
        }

        public Int32? Value
        {
            get
            {
                var val = ((dynamic)element).autoNumeric("get");
                if (Q.IsTrimmedEmpty(val))
                    return null;
                else
                    return Int32.Parse(val, 10);
            }
            set
            {
                if (value == null || value.As<string>() == "")
                    element.Value("");
                else
                    ((dynamic)element).autoNumeric("set", value);
            }
        }
    }

    [Imported, Serializable, Reflectable]
    public class IntegerEditorOptions
    {
        public Int64? MinValue { get; set; }
        public Int64? MaxValue { get; set; }
    }
}

File: Serenity.Scripts/CoreLib/UI/Editors/IntegerEditor.ts
namespace Serenity {

    export interface IntegerEditorOptions {
        minValue?: number;
		maxValue?: number;
		allowNegatives?: boolean;
    }

    @Decorators.registerEditor('Serenity.IntegerEditor', [IDoubleValue])
    @Decorators.element('<input type="text"/>')
    export class IntegerEditor extends Widget<IntegerEditorOptions> implements IDoubleValue {

        constructor(input: JQuery, opt?: IntegerEditorOptions) {
            super(input, opt);

            input.addClass('integerQ');
            var numericOptions = Q.extend(Serenity.DecimalEditor.defaultAutoNumericOptions(),
                {
					vMin: Q.coalesce(this.options.minValue, this.options.allowNegatives ? (this.options.maxValue != null ? ("-" + this.options.maxValue) : '-2147483647') : '0'),
                    vMax: Q.coalesce(this.options.maxValue, 2147483647),
                    aSep: null
                });

            if ($.fn.autoNumeric)
                (input as any).autoNumeric(numericOptions);
        }

        get_value(): number {
            if ($.fn.autoNumeric) {
                var val = (this.element as any).autoNumeric('get') as string;
                if (!!Q.isTrimmedEmpty(val))
                    return null;
                else 
                    return parseInt(val, 10);
            } 
            else {
                var val = Q.trimToNull(this.element.val());
                if (val == null)
                    return null;
                return Q.parseInteger(val)
            }

        }

        get value(): number {
            return this.get_value();
        }

        set_value(value: number) {
            if (value == null || (value as any) === '')
                this.element.val('');
            else if ($.fn.autoNumeric)
                (this.element as any).autoNumeric('set', value);
            else
                this.element.val(Q.formatNumber(value));
        }

        set value(v: number) {
            this.set_value(v);
        }

        get_isValid(): boolean {
            return !isNaN(this.get_value());
        }
    }
}